common/lib/dataset.py

```diff
@@ -811,31 +824,10 @@ def get_columns(self):
 			# no file to get columns from
 			return False
 
-		if self.get_results_path().suffix.lower() == ".csv":
-			with self.get_results_path().open(encoding="utf-8") as infile:
-				own_processor = self.get_own_processor()
-				csv_parameters = own_processor.get_csv_parameters(csv) if own_processor else {}
-
-				reader = csv.DictReader(infile, **csv_parameters)
-				try:
-					return list(reader.fieldnames)
-				except (TypeError, ValueError):
-					# not a valid CSV file?
-					return []
-
-		elif self.get_results_path().suffix.lower() == ".ndjson" and hasattr(self.get_own_processor(), "map_item"):
-			with self.get_results_path().open(encoding="utf-8") as infile:
-				first_line = infile.readline()
-
-			try:
-				item = json.loads(first_line)
-				return list(self.get_own_processor().map_item(item).keys())
-			except (json.JSONDecodeError, ValueError):
-				# not a valid NDJSON file?
-				return []
-
+		if (self.get_results_path().suffix.lower() == ".csv") or (self.get_results_path().suffix.lower() == ".ndjson" and self.get_own_processor() is not None and self.get_own_processor().map_item_method_available(dataset=self)):
```

Comment:
Are the extension checks necessary here? I think if there is a map_item there will be columns and if not then not, regardless of extension

Reply:
(I know they were already there, but I think they might be a left-over from when we were less extension-agnostic)

Reply:
I vaguely remember issue here with something in the frontend. `get_item_keys` would fail because `get_own_processor` would return `None`. But perhaps it the extension checks are redundant. I can test this.

Reply:
CSV datasets do not have `map_item`, but do have columns we wish to return. NDJSON datasets should only return columns if they can be mapped (else keys are not consistent enough to be used as "columns"). Some datasets do not have processors (rare; from what I can tell only deprecated datasets) and a check here avoids an error being raised by `iterate_items`.

Potentially we could combine the two Dataset methods `get_item_keys` and `get_columns`. They are _almost_ the same thing. `get_item_keys` is used in various processors while `get_columns` is used by the frontend and most often `get_options`. Currently, `get_item_keys` takes advantage of `iterate_items` and grabs the first item's keys. `get_columns` was a copy of the code from `iterate_items` which is the code that I'm modifying in this PR.

When attempting to consolidate the code, I realized that `get_columns` could not be completely replaced by `get_item_keys` because of the previously mentioned instances where we do not wish to return columns. Possibly we do not wish to return item keys either in these instances via `get_item_keys`, but I did not explore all usages of that function as no errors were occurring. Mostly likely, `get_columns` returns `False` via the frontend and so the backend never runs into a misuse of `get_item_keys`.